pkgs/development/python-modules/pgvector/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, asyncpg
+, buildPythonPackage
+, django
+, fetchFromGitHub
+, numpy
+, postgresql
+, postgresqlTestHook
+, psycopg
+, psycopg2
+, pytestCheckHook
+, pythonOlder
+, sqlalchemy
+}:
+
+buildPythonPackage rec {
+  pname = "pgvector";
+  version = "0.1.8";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "pgvector";
+    repo = "pgvector-python";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-+0daPPZGVkqfZw0gOZwIl+knL/zZki9fs5kA3dYqPpE=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  nativeCheckInputs = [
+    asyncpg
+    django
+    (postgresql.withPackages (p: with p; [ pgvector ]))
+    postgresqlTestHook
+    psycopg
+    psycopg2
+    pytestCheckHook
```

Comment:
```suggestion
    pytest-asyncio
    pytestCheckHook
```

because currently we are skipping at least one async test case.

```
pgvector> =============================== warnings summary ===============================
pgvector> ../../nix/store/hzv8xjxk35i72jrljvjhl9y5i00vnsqn-python3.10-pytest-7.2.1/lib/python3.10/site-packages/_pytest/config/__init__.py:1294
pgvector>   /nix/store/hzv8xjxk35i72jrljvjhl9y5i00vnsqn-python3.10-pytest-7.2.1/lib/python3.10/site-packages/_pytest/config/__init__.py:1294: PytestConfigWarning: Unknown config option: asyncio_mode
pgvector>   
pgvector>     self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")
pgvector> 
pgvector> tests/test_asyncpg.py:9
pgvector>   /build/source/tests/test_asyncpg.py:9: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
pgvector>     @pytest.mark.asyncio
pgvector> 
pgvector> tests/test_psycopg.py:75
pgvector>   /build/source/tests/test_psycopg.py:75: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
pgvector>     @pytest.mark.asyncio
pgvector> 
pgvector> tests/test_asyncpg.py::TestAsyncpg::test_works
pgvector> tests/test_psycopg.py::TestPsycopg::test_async
pgvector>   /nix/store/hzv8xjxk35i72jrljvjhl9y5i00vnsqn-python3.10-pytest-7.2.1/lib/python3.10/site-packages/_pytest/python.py:184: PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
pgvector>   You need to install a suitable plugin for your async framework, for example:
pgvector>     - anyio
pgvector>     - pytest-asyncio
pgvector>     - pytest-tornasync
pgvector>     - pytest-trio
pgvector>     - pytest-twisted
pgvector>     warnings.warn(PytestUnhandledCoroutineWarning(msg.format(nodeid)))
pgvector> 
pgvector> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

Reply:
Good catch!
I added it.